### DISCUSSION TITLE
How to define a node that accepts a predicate

### DISCUSSION THREAD

**Discussion by Mearman** [breadboard-ai/breadboard#1195] [Q&A]
There are several cases I'd like to know this for, but right now, I'm trying to replicate [`Array.filter()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) for the new utility kit.
With the current tooling, what is the best way to pass a predicate (or predicates) to a node?
And I suppose, in general, how can I specify the signature of the board to be passed as a parameter?

**Comment by aomarks** [comment]
Good question! It sounds like this pattern would be a variation of or wrapper around a call to `code`. `code` isn't yet supported in the new `build` package, but adding it is very high on my todo list, so if you give me a few days I'll come back with something you should be able to work with.

**Reply by Mearman** [reply]
I was thinking that. But then I thought it'd be even better to also be able to define the signature of a board to be passed in then it could be a code node or any other implementation?
